docs/man/man1/lmktfyctl.1
.TH "LMKTFYRNETES" "1" " lmktfyrnetes User Manuals" "Eric Paris" "Jan 2015"  ""


.SH NAME
.PP
lmktfyctl \- lmktfyctl controls the LMKTFYrnetes cluster manager


.SH SYNOPSIS
.PP
\fBlmktfyctl\fP [OPTIONS]


.SH DESCRIPTION
.PP
lmktfyctl controls the LMKTFYrnetes cluster manager.

.PP
Find more information at 
\[la]https://github.com/GoogleCloudPlatform/lmktfyrnetes\[ra].


.SH OPTIONS
.PP
\fB\-\-alsologtostderr\fP=false
    log to standard error as well as files

.PP
\fB\-\-api\-version\fP=""
    The API version to use when talking to the server

.PP
\fB\-a\fP, \fB\-\-auth\-path\fP=""
    Path to the auth info file. If missing, prompt the user. Only used if using https.

.PP
\fB\-\-certificate\-authority\fP=""
    Path to a cert. file for the certificate authority.

.PP
\fB\-\-client\-certificate\fP=""
    Path to a client key file for TLS.

.PP
\fB\-\-client\-key\fP=""
    Path to a client key file for TLS.

.PP
\fB\-\-cluster\fP=""
    The name of the lmktfyconfig cluster to use

.PP
\fB\-\-context\fP=""
    The name of the lmktfyconfig context to use

.PP
\fB\-h\fP, \fB\-\-help\fP=false
    help for lmktfyctl

.PP
\fB\-\-insecure\-skip\-tls\-verify\fP=false
    If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure.

.PP
\fB\-\-lmktfyconfig\fP=""
    Path to the lmktfyconfig file to use for CLI requests.

.PP
\fB\-\-log\_backtrace\_at\fP=:0
    when logging hits line file:N, emit a stack trace

.PP
\fB\-\-log\_dir\fP=""
    If non\-empty, write log files in this directory

.PP
\fB\-\-log\_flush\_frequency\fP=5s
    Maximum number of seconds between log flushes

.PP
\fB\-\-logtostderr\fP=true
    log to standard error instead of files

.PP
\fB\-\-match\-server\-version\fP=false
    Require server version to match client version

.PP
\fB\-\-namespace\fP=""
    If present, the namespace scope for this CLI request.

.PP
\fB\-\-password\fP=""
    Password for basic authentication to the API server.

.PP
\fB\-s\fP, \fB\-\-server\fP=""
    The address and port of the LMKTFYrnetes API server

.PP
\fB\-\-stderrthreshold\fP=2
    logs at or above this threshold go to stderr

.PP
\fB\-\-token\fP=""
    Bearer token for authentication to the API server.

.PP
\fB\-\-user\fP=""
    The name of the lmktfyconfig user to use

.PP
\fB\-\-username\fP=""
    Username for basic authentication to the API server.

.PP
\fB\-\-v\fP=0
    log level for V logs

.PP
\fB\-\-validate\fP=false
    If true, use a schema to validate the input before sending it

.PP
\fB\-\-vmodule\fP=
    comma\-separated list of pattern=N settings for file\-filtered logging


.SH SEE ALSO
.PP
\fBlmktfyctl\-get(1)\fP, \fBlmktfyctl\-describe(1)\fP, \fBlmktfyctl\-create(1)\fP, \fBlmktfyctl\-update(1)\fP, \fBlmktfyctl\-delete(1)\fP, \fBlmktfyctl\-namespace(1)\fP, \fBlmktfyctl\-log(1)\fP, \fBlmktfyctl\-rollingupdate(1)\fP, \fBlmktfyctl\-resize(1)\fP, \fBlmktfyctl\-exec(1)\fP, \fBlmktfyctl\-port\-forward(1)\fP, \fBlmktfyctl\-proxy(1)\fP, \fBlmktfyctl\-run\-container(1)\fP, \fBlmktfyctl\-stop(1)\fP, \fBlmktfyctl\-expose(1)\fP, \fBlmktfyctl\-label(1)\fP, \fBlmktfyctl\-config(1)\fP, \fBlmktfyctl\-clusterinfo(1)\fP, \fBlmktfyctl\-apiversions(1)\fP, \fBlmktfyctl\-version(1)\fP,


.SH HISTORY
.PP
January 2015, Originally compiled by Eric Paris (eparis at redhat dot com) based on the lmktfyrnetes source material, but hopefully they have been automatically generated since!


news.md
---
layout: news
title: News
permalink: /news/
show_in_nav: true
slug: news

hero:
    title: News
    text: Read what others are saying about Kubernetes.
    img: /img/desktop/news/hero_icon.svg

default_img: /img/desktop/news/img_placeholder@2x.jpg

load_more_label: Load More

---


cluster/lmktfyctl.sh
#!/bin/bash

# Copyright 2014 Google Inc. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

set -o errexit
set -o nounset
set -o pipefail

LMKTFY_ROOT=$(dirname "${BASH_SOURCE}")/..
source "${LMKTFY_ROOT}/cluster/lmktfy-env.sh"
UTILS=${LMKTFY_ROOT}/cluster/${LMKTFYRNETES_PROVIDER}/util.sh
if [ -f ${UTILS} ]; then
    source "${UTILS}"
fi

# Get the absolute path of the directory component of a file, i.e. the
# absolute path of the dirname of $1.
get_absolute_dirname() {
  echo "$(cd "$(dirname "$1")" && pwd)"
}

# Detect the OS name/arch so that we can find our binary
case "$(uname -s)" in
  Darwin)
    host_os=darwin
    ;;
  Linux)
    host_os=linux
    ;;
  *)
    echo "Unsupported host OS.  Must be Linux or Mac OS X." >&2
    exit 1
    ;;
esac

case "$(uname -m)" in
  x86_64*)
    host_arch=amd64
    ;;
  i?86_64*)
    host_arch=amd64
    ;;
  amd64*)
    host_arch=amd64
    ;;
  arm*)
    host_arch=arm
    ;;
  i?86*)
    host_arch=x86
    ;;
  *)
    echo "Unsupported host arch. Must be x86_64, 386 or arm." >&2
    exit 1
    ;;
esac

# If LMKTFYCTL_PATH isn't set, gather up the list of likely places and use ls
# to find the latest one.
if [[ -z "${LMKTFYCTL_PATH:-}" ]]; then
  locations=(
    "${LMKTFY_ROOT}/_output/dockerized/bin/${host_os}/${host_arch}/lmktfyctl"
    "${LMKTFY_ROOT}/_output/local/bin/${host_os}/${host_arch}/lmktfyctl"
    "${LMKTFY_ROOT}/platforms/${host_os}/${host_arch}/lmktfyctl"
  )
  lmktfyctl=$( (ls -t "${locations[@]}" 2>/dev/null || true) | head -1 )

  if [[ ! -x "$lmktfyctl" ]]; then
    {
      echo "It looks as if you don't have a compiled lmktfyctl binary"
      echo
      echo "If you are running from a clone of the git repo, please run"
      echo "'./build/run.sh hack/build-cross.sh'. Note that this requires having"
      echo "Docker installed."
      echo
      echo "If you are running from a binary release tarball, something is wrong. "
      echo "Look at http://lmktfy.io/ for information on how to contact the "
      echo "development team for help."
    } >&2
    exit 1
  fi
elif [[ ! -x "${LMKTFYCTL_PATH}" ]]; then
  {
    echo "LMKTFYCTL_PATH environment variable set to '${LMKTFYCTL_PATH}', but "
    echo "this doesn't seem to be a valid executable."
  } >&2
  exit 1
fi
lmktfyctl="${LMKTFYCTL_PATH:-${lmktfyctl}}"

# While GKE requires the lmktfyctl binary, it's actually called through
# gcloud. But we need to adjust the PATH so gcloud gets the right one.
if [[ "$LMKTFYRNETES_PROVIDER" == "gke" ]]; then
  detect-project &> /dev/null
  export PATH=$(get_absolute_dirname $lmktfyctl):$PATH
  lmktfyctl="${GCLOUD}"
  # GKE runs lmktfyctl through gcloud.
  config=(
    "preview"
    "container"
    "lmktfyctl"
    "--project=${PROJECT}"
    "--zone=${ZONE}"
    "--cluster=${CLUSTER_NAME}"
  )
elif [[ "$LMKTFYRNETES_PROVIDER" == "vagrant" ]]; then
  # When we are using vagrant it has hard coded lmktfyconfig, and do not clobber public endpoints
  config=(
    "--lmktfyconfig=$HOME/.lmktfy_vagrant_lmktfyconfig"
  )
elif [[ "$LMKTFYRNETES_PROVIDER" == "libvirt-coreos" ]]; then
  detect-master > /dev/null
  config=(
    "--server=http://${LMKTFY_MASTER_IP}:8080"
  )
fi

echo "current-context: \"$(${lmktfyctl} "${config[@]:+${config[@]}}" config view -o template --template='{{index . "current-context"}}')\"" >&2

echo "Running:" "${lmktfyctl}" "${config[@]:+${config[@]}}" "${@+$@}" >&2
"${lmktfyctl}" "${config[@]:+${config[@]}}" "${@+$@}"
